### PR TITLE
Us006 Filters

### PR DESCRIPTION
**Pull request by kuzivka**
Added buttons for filtering

### CODE REVIEW

File: src/components/FilterBar.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  const dispatch = useDispatch();
+
+  const filterState = useSelector(getFilter);
+
+  const handleDeleteAllClick = useCallback(() => {
+    dispatch(deleteComletedFromTodoList());
+    dispatch(filterTodoList('all'));
+  }, [dispatch]);
+
+  const buttons: TaskToShow[] = ['all', 'completed', 'active'];
```
Comment (Yuliia143) [inline src/components/FilterBar.tsx]: Try to work with enums for 'all', 'completed', 'active', create it in separate file and use

File: src/components/TaskCards.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
-import { List } from '@mui/material';
-import { useState, useCallback } from 'react';
+import { List, Typography } from '@mui/material';
+import { useState, Fragment, useCallback } from 'react';
```
Comment (Yuliia143) [inline src/components/TaskCards.tsx]: Why do you need Fragment again ?

File: src/components/TaskCards.tsx [hunk 3]
```diff
@@ -27,6 +27,9 @@ export default function TaskCards() {
           onClose={resetTaskId}
         />
       )}
+      {tasks.length === 0 && (
```
Comment (Yuliia143) [inline src/components/TaskCards.tsx]: !task.length ?

File: src/components/styles/style.scss [hunk 4]
```diff
@@ -38,6 +38,18 @@
...
   justify-content: center;
 }
+
+.no-item-title {
+  color: #ddd;
+}
+.sort-icon{
+  width: 70px !important;
+  font-size: 60px !important;
+  // border: 1px solid black;
```
Comment (Yuliia143) [inline src/components/styles/style.scss]: Commented code

File: src/reducers/todoReducer.ts [hunk 5]
```diff
@@ -52,6 +62,14 @@ export const todoReducer = (
       const newState = { ...state, tasks: updatedTasks };
       return newState;
     }
+    case FILTER_TODO: {
+      const newState: ToDoState = { ...state, filterBy: action.payload };
+      return newState;
+    }
+    case DELETE_COMPLETED: {
+      const newListOfTasks = state.tasks.filter((task) => task.done === false);
```
Comment (Yuliia143) [inline src/reducers/todoReducer.ts]: === false

File: src/selectors/getTaskListToShowSelector.ts [hunk 6]
```diff
@@ -1,11 +1,22 @@
 import { Selector } from '@reduxjs/toolkit';
 import { format } from 'date-fns';
-import { TIME_ON_CARDS } from '../components/constants';
+import { TIME_ON_CARDS } from '../constants';
+
+const applyFiltering = (filterBy: TaskToShow) => (task: ToDo) => {
+  if (filterBy === 'active') {
+    return !task.done;
+  }
+  if (filterBy === 'completed') {
```
Comment (Yuliia143) [inline src/selectors/getTaskListToShowSelector.ts]: use enums

**Review by Yuliia143 — CHANGES_REQUESTED** [review]

File: src/components/styles/style.scss (hunk 4 above)
Comment (kuzivka) [reply src/components/styles/style.scss]: fixed

File: src/reducers/todoReducer.ts (hunk 5 above)
Comment (kuzivka) [reply src/reducers/todoReducer.ts]: fixed

File: src/enums.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+export enum filterListBy {
+    all = 'all',
+    comleted = 'completed',
+    active = 'active',
+  }
+
```
Comment (Yuliia143) [inline src/enums.ts]: No newline at end of file

File: src/components/EditTodoModal.tsx [hunk 8]
```diff
@@ -27,6 +27,7 @@ export function EditTodoModal(props: EditTodoModalProps) {
           done: false,
         })
       );
+      dispatch(filterTodoList('all'));
```
Comment (Yuliia143) [inline src/components/EditTodoModal.tsx]: 'all' ?

File: src/components/AddNewTodoModal.tsx [hunk 9]
```diff
@@ -16,6 +16,7 @@ export function AddNewTodoModal(props: IAddNewTodoModalProps) {
   const handleSave = useCallback(
     (taskValue: string, expirationDate: number) => {
       dispatch(addTodo(getNewTaskObject(taskValue, expirationDate.valueOf())));
+      dispatch(filterTodoList('all'));
```
Comment (Yuliia143) [inline src/components/AddNewTodoModal.tsx]: 'all'?

**Review by Yuliia143 — APPROVED** [review]

**Pull request merged**